lib/ansible/modules/cloud/ovirt/ovirt_vms.py

```diff
@@ -1243,6 +1250,48 @@ def main():
             )
         elif state == 'absent':
             ret = vms_module.remove()
+        elif state == 'registered':
+            storage_domains_service = connection.system_service().storage_domains_service()
+
+            # Find the storage domain with unregistered VM:
+            sd_id = get_id_by_name(storage_domains_service, module.params['storage_domain'])
+            storage_domain_service = storage_domains_service.storage_domain_service(sd_id)
+            vms_service = storage_domain_service.vms_service()
+
+            # Find the the unregistered VM we want to register:
+            vms = vms_service.list(query={'unregistered': 'true'})
+            vm = next(
```

Comment:
This looks strange, I think you can just use:

In general:

1) if VM is already in system, you should just return that template, currently it's tryting to fetch that template from storage domain rather then system.

2) If VM is not in system, but it's in storage domain as unregistered, we should run `register` action. 

3) I think we should support both `id` and `name`.